Detectors/TPC/spacecharge/include/TPCSpaceCharge/SpaceChargeHelpers.h

```diff
@@ -0,0 +1,269 @@
+/// \date Aug 21, 2020
+
+#ifndef ALICEO2_TPC_SPACECHARGEHELPERS_H_
+#define ALICEO2_TPC_SPACECHARGEHELPERS_H_
+
+#include <functional>
+#include <cmath>
+#include "TPCSpaceCharge/TriCubic.h"
+#include "DataFormatsTPC/Defs.h"
+
+namespace o2
+{
+namespace tpc
+{
+
+///
+/// this class contains an analytical description of the space charge, potential and the electric fields.
+/// The analytical functions can be used to test the poisson solver and the caluclation of distortions/corrections.
+///
+template <typename DataT = double>
+class AnalyticalFields
+{
+ public:
+  AnalyticalFields(const o2::tpc::Side side = o2::tpc::Side::A) : mSide{side} {};
+
+  o2::tpc::Side getSide() const { return mSide; }
+
+  void setSide(const o2::tpc::Side side) { mSide = side; }
+
+  /// sets the parameters
+  void setParameters(const DataT parA, const DataT parB, const DataT parC)
+  {
+    mParA = parA;
+    mParB = parB;
+    mParC = parC;
+  }
+
+  /// return parameter A
+  DataT getParA() const { return mParA; }
+
+  /// return parameter B
+  DataT getParB() const { return mParB; }
+
+  /// return parameter C
+  DataT getParC() const { return mParC; }
+
+  /// \param r r coordinate
+  /// \param phi phi coordinate
+  /// \param z z coordinate
+  /// \return returns the function value for electric field Er for given coordinate
+  DataT evalEr(DataT z, DataT r, DataT phi) const { return mErFunc(z, r, phi); }
+
+  /// \param r r coordinate
+  /// \param phi phi coordinate
+  /// \param z z coordinate
+  /// \return returns the function value for electric field Ez for given coordinate
+  DataT evalEz(DataT z, DataT r, DataT phi) const { return mEzFunc(z, r, phi); }
+
+  /// \param r r coordinate
+  /// \param phi phi coordinate
+  /// \param z z coordinate
+  /// \return returns the function value for electric field Ephi for given coordinate
+  DataT evalEphi(DataT z, DataT r, DataT phi) const { return mEphiFunc(z, r, phi); }
+
+  /// \param r r coordinate
+  /// \param phi phi coordinate
+  /// \param z z coordinate
+  /// \return returns the function value for the potential for given coordinate
+  DataT evalPotential(DataT z, DataT r, DataT phi) const { return mPotentialFunc(z, r, phi); }
+
+  /// \param r r coordinate
+  /// \param phi phi coordinate
+  /// \param z z coordinate
+  /// \return returns the function value for the space charge density for given coordinate
+  DataT evalDensity(DataT z, DataT r, DataT phi) const { return mDensityFunc(z, r, phi); }
+
+  /// analytical potential
+  std::function<DataT(DataT, DataT, DataT)> mPotentialFunc = [& mParA = mParA, &mParB = mParB, &mParC = mParC](const DataT z, const DataT r, const DataT phi) {
+    const DataT zz = std::abs(z);
+    return -mParA * (std::pow((-r + 254.5 + 83.5), 4) - 338.0 * std::pow((-r + 254.5 + 83.5), 3) + 21250.75 * std::pow((-r + 254.5 + 83.5), 2)) * std::cos(mParB * phi) * std::cos(mParB * phi) * std::exp(-1 * mParC * (zz - 125) * (zz - 125));
```

Comment:
depending on where or how often this function is used during runtime, I'd like to point out that constructs  like `std::pow((-r + 254.5 + 83.5), 4)` are unfortunately inducing a function call to pow and therefore slow. You can check with https://godbolt.org/ that this is only optimized away with `-ffast-math` which we are not using.

Reply:
thanks for the comment :) I didnt knew that about `std::pow`. But since these functions are only used for "debugging/testing" anyway, it does not really matter how fast the evaluation is.